src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -2050,6 +2050,175 @@ void TypeObjectUtils::plain_collection_type_identifier_header_consistency(
+        }
+    }
+    else
+    {
+        return EK_BOTH;
+    }
+    return TK_NONE;
+}
+
+void TypeObjectUtils::plain_map_type_identifier_header_consistency(
+        const PlainCollectionHeader& header,
+        const TypeIdentifier& element_identifier,
+        const TypeIdentifier& key_identifier)
+{
+    EquivalenceKind key_equiv_kind = get_map_component_equiv_kind_for_consistency(key_identifier, true);
+    EquivalenceKind element_equiv_kind = get_map_component_equiv_kind_for_consistency(element_identifier, false);
+
+    switch (header.equiv_kind())
+    {
+        case EK_BOTH:
+            if (key_equiv_kind != EK_BOTH)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_BOTH, but the key_identifier is not fully descriptive.");
+            }
+            if (element_equiv_kind != EK_BOTH)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_BOTH, but the element_identifier is not fully descriptive.");
+            }
+            break;
+        case EK_MINIMAL:
+            if (key_equiv_kind == EK_BOTH &&
+                    element_equiv_kind == EK_BOTH)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_MINIMAL, but both identifiers are fully descriptive.");
+            }
+
+            if (element_equiv_kind != EK_BOTH && element_equiv_kind != EK_MINIMAL)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_MINIMAL, but the element_identifier is neither fully descriptive nor EK_MINIMAL.");
+            }
+
+            if (key_equiv_kind != EK_BOTH && key_equiv_kind != EK_MINIMAL)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_MINIMAL, but the key_identifier is neither fully descriptive nor EK_MINIMAL.");
+            }
+            break;
+        case EK_COMPLETE:
+            if (key_equiv_kind == EK_BOTH &&
+                    element_equiv_kind == EK_BOTH)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_COMPLETE, but both identifiers are fully descriptive.");
+            }
+
+            if (element_equiv_kind != EK_BOTH && element_equiv_kind != EK_COMPLETE)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_COMPLETE, but the element_identifier is neither fully descriptive nor EK_COMPLETE.");
+            }
+
+            if (key_equiv_kind != EK_BOTH && key_equiv_kind != EK_COMPLETE)
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_COMPLETE, but the key_identifier is neither fully descriptive nor EK_COMPLETE.");
+            }
+            break;
+    }
+}
```

Comment:
I have the feeling that this is not necessary. You already have created a method which returns the EquivalenceKind, so I think it would be enough to compare the EquivalenceKind of the header with the one returned by the other method when provided with the map TypeIdentifier, don't you think?

Reply:
changed in [a989bd3](https://github.com/eProsima/Fast-DDS/pull/4496/commits/a989bd32df198abf53da0faff2c924f05a240c47)